File: UI/src/features/game/models/PublicGameState.ts
import type { HandType } from "./HandType";
import type { Player } from "./Player";
import type { Rank } from "./Rank";
import type { Suit } from "./Suit";

export interface PublicGameState {
    hasStarted: boolean;
    players: { [key: string]: Player; };
    playerOrder: string[];
    currentTurnPlayerId: string | null;
    currentClaimedHand: HandType | null;
    claimingPlayerId: string | null;
    ranks: Rank[] | null;
    claimedSuit: Suit | null;
    maxCardCount: number;
}

File: WebApi.Common/Features/Games/Models/PublicGameState.cs
using WebApi.Common.Features.Players.Models;

namespace WebApi.Common.Features.Games.Models;

public record PublicGameState(
    string GameCode,
    bool HasStarted,
    IReadOnlyDictionary<string, Player> Players,
    IReadOnlyList<string> PlayerOrder,
    string? CurrentTurnPlayerId,
    HandType? CurrentClaimedHand,
    string? ClaimingPlayerId,
    List<Rank>? Ranks,
    Suit? ClaimedSuit,
    int MaxCardCount,
    IReadOnlyList<RoundHistoryEntry> RoundHistory
)
{
    public static PublicGameState FromGameState(GameState gameState) => new(
        gameState.GameCode,
        gameState.HasStarted,
        gameState.Players,
        gameState.PlayerOrder,
        gameState.CurrentTurnPlayerId,
        gameState.CurrentClaimedHand,
        gameState.ClaimingPlayerId,
        gameState.Ranks,
        gameState.ClaimedSuit,
        gameState.MaxCardCount,
        gameState.RoundHistory
    );
}
